File: python/wordy.py
import re

def answer(question):
    arr = re.split(r'(-?\d+|plus|minus|multiplied by|divided by)', question)
    
    st = []
    ops = []
    is_op = False
    
    for each in arr:
        if each in ['What is ', '?', ' ', 'What is?']:
            continue
        
        if re.search(r'-?\d+', each):
            if is_op:
                raise ValueError('syntax error')
            st.append(int(each))
            is_op = not is_op
        elif each in ['plus', 'minus', 'multiplied by', 'divided by']:
            if not is_op:
                raise ValueError('syntax error')
            ops.append(each)
            is_op = not is_op
        else:
            raise ValueError('unknown operation')

    if st == []:
        raise ValueError('syntax error')
        
    st = st[::-1]
    for op in ops:
        if len(st) < 2:
            raise ValueError('syntax error')
        
        a = st.pop()
        b = st.pop()
        
        if op == 'plus':
            st.append(a + b)
        elif op == 'minus':
            st.append(a - b)
        elif op == 'multiplied by':
            st.append(a * b)
        else:
            st.append(a // b)
            
    if len(st) != 1:
        raise ValueError('syntax error')
    
    return st[0]

File: go/wordy.go
package wordy

import (
	"strconv"
	"strings"
)

func Answer(question string) (int, bool) {
	question = strings.Trim(question, "?What is")
	question = strings.ReplaceAll(question, "multiplied by", "multiplied")
	question = strings.ReplaceAll(question, "divided by", "divided")
	arr := strings.Split(question, " ")
	if len(arr) == 0 || len(arr)%2 == 0 {
		return 0, false
	}
	ans, err := strconv.Atoi(arr[0])
	if err != nil {
		return 0, false
	}

	for i := 1; i < len(arr); i += 2 {
		n, err := strconv.Atoi(arr[i+1])
		if err != nil {
			return 0, false
		}
		switch arr[i] {
		case "plus":
			ans += n
		case "minus":
			ans -= n
		case "divided":
			ans /= n
		case "multiplied":
			ans *= n
		default:
			return 0, false
		}
	}
	return ans, true
}
